js/delta.js

```diff
@@ -376,76 +376,102 @@ module.exports = class delta extends Exchange {
+            const settlingAsset = this.safeValue (market, 'settling_asset');
             const baseId = this.safeString (underlyingAsset, 'symbol');
             const quoteId = this.safeString (quotingAsset, 'symbol');
+            const settleId = this.safeString (settlingAsset, 'symbol');
             const id = this.safeString (market, 'symbol');
             const numericId = this.safeInteger (market, 'id');
             const base = this.safeCurrencyCode (baseId);
             const quote = this.safeCurrencyCode (quoteId);
+            const settle = this.safeCurrencyCode (settleId);
             let symbol = id;
-            let swap = false;
-            let future = false;
-            let option = false;
-            if (type === 'perpetual_futures') {
+            const callOptions = (type === 'call_options');
+            const putOptions = (type === 'put_options');
+            const moveOptions = (type === 'move_options');
+            const spot = (type === 'spot');
+            const swap = (type === 'perpetual_futures');
+            const future = (type === 'futures');
+            const option = (callOptions || putOptions || moveOptions);
+            const contract = (swap || future || option);
+            const strike = this.safeNumber (market, 'strike_price');
+            const expiryDatetime = this.safeString (market, 'settlement_time');
+            const expiry = this.parse8601 (expiryDatetime);
+            const contractSize = this.safeNumber (market, 'contract_value');
+            const linear = (settle === base);
+            let optionType = undefined;
+            if (swap) {
                 type = 'swap';
-                swap = true;
-                future = false;
-                option = false;
-                if (id.indexOf ('_') < 0) {
-                    symbol = base + '/' + quote;
-                }
-            } else if ((type === 'call_options') || (type === 'put_options') || (type === 'move_options')) {
-                type = 'option';
-                swap = false;
-                option = true;
-                future = false;
-            } else if (type === 'futures') {
+                symbol = base + '/' + quote + ':' + settle;
+            } else if (future) {
                 type = 'future';
-                swap = false;
-                option = false;
-                future = true;
+                symbol = base + '/' + quote + ':' + settle + '-' + this.yymmdd (expiry);
+            } else if (option) {
+                type = 'option';
+                let letter = 'C';
+                optionType = 'call';
+                if (putOptions) {
+                    letter = 'P';
+                    optionType = 'put';
+                } else if (moveOptions) {
+                    letter = 'M';
+                    optionType = 'move';
+                }
+                symbol = base + '/' + quote + ':' + settle + '-' + this.yymmdd (expiry) + ':' + strike + ':' + letter;
```

Comment:
The `strike` price must be a string here, we should use safeString for concatenation, and then `parseNumber (strike)` upon filling up the market.